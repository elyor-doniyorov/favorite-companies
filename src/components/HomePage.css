.homepage-logo {
  background-color: #d1447a;
}

.homepage-logo-text {
  text-align: right;
  color: white;
  padding-top: 70px;
  line-height: 0%;
}

.homepage-logo-img {
  padding-left: 30px;
  margin-left: 100px;
  padding-top: 30px;
  margin-bottom: 20px;
  width: 150px;
  opacity: 0.3;
}

.top {
  font-style: italic;
}

.seller-container {
  box-sizing: border-box;
}

.seller-info {
  height: 200px;
}

.home-sub-title {
  padding-left: 30px;
  color: white;
  background-color: #db4681;
}
