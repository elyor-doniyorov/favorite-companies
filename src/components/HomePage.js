import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchSellers } from '../redux/home/home';
import Seller from './Seller';
import './HomePage.css';

const HomePage = () => {
  const sellers = useSelector((state) => state.homeReducer);
  const dispatch = useDispatch();
  useEffect(() => {
    if (!sellers.length > 0) {
      dispatch(fetchSellers());
    }
  }, []);
  return (
    <div>
      <div className="row homepage-logo">
        <div className="col-6">
          <img
            className="homepage-logo-img"
            src="/images/seller.jpeg"
            alt="logo"
          />
        </div>
        <div className="col-6 homepage-logo-text">
          <h2>Visit Big sellers online</h2>
          <p className="top">Top 100</p>
        </div>
      </div>
      <div className="row home-sub-title">START BY SELLER</div>
      <div className="seller-container row">
        {sellers.map((seller) => (
          <Seller key={seller.symbol} seller={seller} />
        ))}
      </div>
    </div>
  );
};

export default HomePage;
