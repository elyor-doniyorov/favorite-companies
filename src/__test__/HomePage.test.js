import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import HomePage from '../components/HomePage';
import store from '../redux/configStore';

describe('check if all component are rendered', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><HomePage /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
