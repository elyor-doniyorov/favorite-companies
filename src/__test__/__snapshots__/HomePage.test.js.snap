// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if all component are rendered renders correctly 1`] = `
<div>
  <div
    className="row homepage-logo"
  >
    <div
      className="col-6"
    >
      <img
        alt="logo"
        className="homepage-logo-img"
        src="/images/favorite.png"
      />
    </div>
    <div
      className="col-6 homepage-logo-text"
    >
      <h2>
        Visit Top Companies online
      </h2>
      <p
        className="top"
      >
        Top 100
      </p>
    </div>
  </div>
  <div
    className="row home-sub-title"
  >
    START BY SELLER
  </div>
  <div
    className="seller-container row"
  />
</div>
`;
